OpenSim/Common/XsensDataReader.cpp

```diff
@@ -62,13 +63,20 @@ XsensDataReader::extendRead(const std::string& folderName) const {
                     dataRate = std::stod(tokens[3]);
                 }
             }
-            if (j == labels_line_number) { // Find indices for PacketCounter, Acc_{X,Y,Z}, Gyr_{X,Y,Z}, Mag_{X,Y,Z} on line 5
+            if (!labelsFound) { // Find indices for PacketCounter, Acc_{X,Y,Z}, Gyr_{X,Y,Z}, Mag_{X,Y,Z} on line 5
```

Comment:
The `labelsFound` and `packetCounterIndex != -1` seem redundant. I don't see how this `if` is doing anything since `labelsFound` must be false to increment the `for` loop. I would expect the condition to be something like `while(packetCounterIndex < 0)` and a comment about incrementing the line number until the packet counter is found.

Reply:
While I think labelsFound is more readable, and scales better to other methods of detecting labels than checking a index==-1, I made the suggested change anyway since it's rather local  in the function implementation.